### PR TITLE
Varnish support

### PR DESCRIPTION
**Pull request by Jancis**
# PR is blocked by https://github.com/wunderio/silta-images/pull/172, update image tag in this PR

This PR includes:
- Varnish support for frontend chart
- nginx configmap newline fixes
- nginx parameters `port_in_redirect` and `server_name_in_redirect` to fix redirects;

### CODE REVIEW

File: charts/frontend/values.yaml [hunk 1]
```diff
@@ -378,6 +378,53 @@ mariadb:
...
 
+varnish:
+  enabled: false
+  resources:
+    requests:
+      cpu: 25m
+      memory: 32Mi
+  image: wunderio/silta-varnish
+  # TODO: change to 7-v1
+  imageTag: 7-test
```
Comment (Rade333) [inline charts/frontend/values.yaml:389]: Just adding this as a comment so it's not forgotten: Todo change image tag.

File: charts/frontend/templates/varnish-deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+      service: varnish
+  template:
+    metadata:
+      labels:
+        {{- include "frontend.release_labels" . | nindent 8 }}
+        service: varnish
+        silta-ingress: allow
+      annotations:
+        # We use a checksum to redeploy the pods when the configMap changes.
+        configMap-checksum: {{ include (print $.Template.BasePath "/varnish-configmap-vcl.yaml") . | sha256sum }}
```
Comment (Rade333) [inline charts/frontend/templates/varnish-deployment.yaml:23]: What about checksum of varnish-secret.yaml, should be added here?

**Review by Rade333 — CHANGES_REQUESTED** [review]
Mostly looking good, check small comments

File: charts/frontend/templates/varnish-deployment.yaml (hunk 2 above)
Comment (Jancis) [reply charts/frontend/templates/varnish-deployment.yaml:23]: good idea!

**Review by Rade333 — APPROVED** [review]

File: charts/frontend/values.yaml (hunk 1 above)
Comment (Rade333) [reply charts/frontend/values.yaml]: Done

**Pull request merged**